lib/client/mfa/cli.go

```diff
@@ -188,12 +222,72 @@ func (c *CLIPrompt) promptWebauthn(ctx context.Context, chal *proto.MFAAuthentic
 }
 
 func (c *CLIPrompt) promptDevicePrefix() string {
-	if c.cfg.DeviceType != "" {
-		return fmt.Sprintf("*%s* ", c.cfg.DeviceType)
+	if c.DeviceType != "" {
+		return fmt.Sprintf("*%s* ", c.DeviceType)
 	}
 	return ""
 }
 
+func (c *CLIPrompt) promptWebauthnAndOTP(ctx context.Context, chal *proto.MFAAuthenticateChallenge) (*proto.MFAAuthenticateResponse, error) {
```

Comment:
Pushing the goroutines only to here is interesting, but it's also a downside in that this gets exercised less often. That is why I always did the goroutines, regardless of the settings.

Reply:
Now that we have unit tests covering this path, I think it's less of a concern, but if you disagree I can revert this. It just doesn't make as much since with SSO since SSO is always on its own.

Reply:
No need to revert.